model_compression_toolkit/qunatizers_infrastructure/keras/quantize_wrapper.py

```diff
@@ -0,0 +1,139 @@
+# Copyright 2022 Sony Semiconductor Israel, Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================f
+
+
+from model_compression_toolkit.core.common.constants import FOUND_TF
+from model_compression_toolkit.core.common.logger import Logger
+from model_compression_toolkit.qunatizers_infrastructure.common.node_quantization_dispatcher import \
+    NodeQuantizationDispatcher
+
+if FOUND_TF:
+    import tensorflow as tf
+    from tensorflow.python.util import tf_inspect
+    from tensorflow_model_optimization.python.core.keras import utils
+    from keras.utils import deserialize_keras_object, serialize_keras_object
+
+
+    def _make_quantizer_fn(quantizer, x, training):
+        """Use currying to return True/False specialized fns to the cond."""
+
+        def quantizer_fn():
+            return quantizer(x, training)
+
+        return quantizer_fn
+
+
+    def _weight_name(name):
+        """Extracts the weight name from the full TensorFlow variable name.
+
+        For example, returns 'kernel' for 'dense_2/kernel:0'.
+
+        Args:
+          name: TensorFlow variable name.
+
+        Returns:
+          Extracted weight name.
+        """
+        return name.split(':')[0].split('/')[-1]
+
+
+    class KerasQuantizationWrapper(tf.keras.layers.Wrapper):
+        def __init__(self,
+                     layer,
+                     dispatcher: NodeQuantizationDispatcher):
+            """
+            Keras Quantization Wrapper takes a keras layer and dispatcher and infer a quantized layer.
+
+            Args:
+                layer: A keras layer.
+                dispatcher: A node quantization dispatcher.
+            """
+            super(KerasQuantizationWrapper, self).__init__(layer)
+            self.dispatcher = dispatcher
+
+        def get_config(self):
+            base_config = super(KerasQuantizationWrapper, self).get_config()
+            config = {'dispatcher': serialize_keras_object(self.dispatcher)}
```

Comment:
simpler to init the config with the dispather, then update it with the super's config, and return it